Submitted By:            Joe Locash  <jlocash at gmail dot com>
Date:                    2026-08-04
Initial Package Version: 1.28.5
Origin:                  Upstream (MR-12217)
Upstream Status:         Merged
Description:             Fixes building with ffmpeg-9

From be9bdb67f4fc868977f1f74e95b6ebe19ef792ec Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 4 Aug 2026 12:09:11 -0500
Subject: [PATCH] gst-libav: Fix build failure with FFmpeg 9.0

Closes https://gitlab.freedesktop.org/gstreamer/gstreamer/-/work_items/5251

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/12217>
---
 subprojects/gst-libav/ext/libav/gstavviddec.c | 15 +++++++++------
 subprojects/gst-libav/ext/libav/gstavvidenc.c | 15 +++++++++------
 2 files changed, 18 insertions(+), 12 deletions(-)

diff --git a/subprojects/gst-libav/ext/libav/gstavviddec.c b/subprojects/gst-libav/ext/libav/gstavviddec.c
index e21f3e15585..39335d98b1c 100644
--- a/subprojects/gst-libav/ext/libav/gstavviddec.c
+++ b/subprojects/gst-libav/ext/libav/gstavviddec.c
@@ -2997,12 +2997,15 @@ gst_ffmpegviddec_register (GstPlugin * plugin)
     }
 
     /* no quasi codecs, please */
-    if (in_plugin->id == AV_CODEC_ID_RAWVIDEO ||
-        in_plugin->id == AV_CODEC_ID_V210 ||
-        in_plugin->id == AV_CODEC_ID_V210X ||
-        in_plugin->id == AV_CODEC_ID_V308 ||
-        in_plugin->id == AV_CODEC_ID_V408 ||
-        in_plugin->id == AV_CODEC_ID_V410 || in_plugin->id == AV_CODEC_ID_R210
+    if (in_plugin->id == AV_CODEC_ID_RAWVIDEO
+        || in_plugin->id == AV_CODEC_ID_V210
+        || in_plugin->id == AV_CODEC_ID_V210X
+        || in_plugin->id == AV_CODEC_ID_R210
+#if LIBAVCODEC_VERSION_MAJOR < 63
+        || in_plugin->id == AV_CODEC_ID_V308
+        || in_plugin->id == AV_CODEC_ID_V408
+        || in_plugin->id == AV_CODEC_ID_V410
+#endif
 #if LIBAVCODEC_VERSION_MAJOR < 61
         || in_plugin->id == AV_CODEC_ID_AYUV
 #endif
diff --git a/subprojects/gst-libav/ext/libav/gstavvidenc.c b/subprojects/gst-libav/ext/libav/gstavvidenc.c
index 2ae6a1b4517..b8e010e70cf 100644
--- a/subprojects/gst-libav/ext/libav/gstavvidenc.c
+++ b/subprojects/gst-libav/ext/libav/gstavvidenc.c
@@ -1030,12 +1030,15 @@ gst_ffmpegvidenc_register (GstPlugin * plugin)
     }
 
     /* no quasi codecs, please */
-    if (in_plugin->id == AV_CODEC_ID_RAWVIDEO ||
-        in_plugin->id == AV_CODEC_ID_V210 ||
-        in_plugin->id == AV_CODEC_ID_V210X ||
-        in_plugin->id == AV_CODEC_ID_V308 ||
-        in_plugin->id == AV_CODEC_ID_V408 ||
-        in_plugin->id == AV_CODEC_ID_V410 || in_plugin->id == AV_CODEC_ID_R210
+    if (in_plugin->id == AV_CODEC_ID_RAWVIDEO
+        || in_plugin->id == AV_CODEC_ID_V210
+        || in_plugin->id == AV_CODEC_ID_V210X
+        || in_plugin->id == AV_CODEC_ID_R210
+#if LIBAVCODEC_VERSION_MAJOR < 63
+        || in_plugin->id == AV_CODEC_ID_V308
+        || in_plugin->id == AV_CODEC_ID_V408
+        || in_plugin->id == AV_CODEC_ID_V410
+#endif
 #if LIBAVCODEC_VERSION_MAJOR < 61
         || in_plugin->id == AV_CODEC_ID_AYUV
 #endif
-- 
GitLab

